source/Nevermore/RelationalStore.cs

```diff
@@ -71,24 +88,31 @@ public class RelationalStore : IRelationalStore
             JsonSerializerSettings jsonSettings,
             IRelatedDocumentStore relatedDocumentStore,
             int keyBlockSize = 20,
-            ObjectInitialisationOptions objectInitialisationOptions = ObjectInitialisationOptions.None)
+            ObjectInitialisationOptions objectInitialisationOptions = ObjectInitialisationOptions.None,
+            bool enableMarsByDefault = true)
         {
             this.registry = new Lazy<RelationalTransactionRegistry>(
-                () => SetConnectionStringOptions(connectionString(), applicationName)
+                () => SetConnectionStringOptions(connectionString(), applicationName, false)
+            );
+
+            this.registryWithMarsOn = new Lazy<RelationalTransactionRegistry>(
+                () => SetConnectionStringOptions(connectionString(), applicationName, true)
             );
+
             this.sqlCommandFactory = sqlCommandFactory;
             this.mappings = mappings;
             keyAllocator = new KeyAllocator(this, keyBlockSize);
 
             this.jsonSettings = jsonSettings;
             this.relatedDocumentStore = relatedDocumentStore;
             this.objectInitialisationOptions = objectInitialisationOptions;
+            this.enableMarsByDefault = enableMarsByDefault;
         }
 
-        public string ConnectionString => registry.Value.ConnectionString;
-        public int MaxPoolSize => registry.Value.MaxPoolSize;
+        public string ConnectionString => SelectRegistry().ConnectionString;
```

Comment:
This will only load the default one tho. It doesn't reflect the actual one in use and after having a closer look of this implementation, I believe initially this was designed to hold a **store level** registry and this registry is meant to have some sort of monitoring/logging helper ability of current used transactions in order to track the usage of the transaction/connection pool or gain the visibility. Thus, having two registries here will break the existing behavior if someone is using it as swapping between two registries will not reflect the actual number of transactions/connections in use in the store. e.g. `WriteCurrentTransactions(StringBuilder sb)` is broken now and each registry holds its own collection of transactions as well as its own sql connection pool due to different connection strings.

Reply:
There are two connections pools so having two registries actually makes a lot of sense. Each connection pool has its own limit.

That being said there is a bug here because `WriteCurrentTransactions` should use both registries.

Reply:
Yeah, and the `MaxPoolSize` and `ConnectionString` property of `RelationalStore` are no longer valid as well. This `RelationalStore` was designed to be like **one-to-one** relationship between itself and `Connection Pool` as it has direct dependency on one single **connection string**.